Update API docs and support guarantees

kashms opened the pull request:
Cleaning up the docs about API types and their support guarantees.

@jason-ha @DLehenbauer @rohandubal @CraigMacomber please review

kashms committed:
Update API docs and support guarantees

Cleaning up the docs about API types and their support guarantees.

kashms committed:
Fixing lint errors

jason-ha commented on docs/content/docs/build/releases-and-apitags.md:
```diff
@@ -37,10 +37,18 @@ There are no guarantees for API breakages or Fluid document compatibility among
+`@fluid-experimental` APIs are for developers to try experimental features and provide feedback. It's possible that these APIs can get completely scrapped or drastically changed in a minor release based on developer feedback.
+`@fluid-internal` APIs are only meant for internal consumption by the framework and should not be used.
+
+For packages that are part of the **@fluidframework** scope, we use import paths to communicate the stability and guarantees associated with those APIs.
```
Is this meant for 1p or general? General we only care about `fluid-framework` (which is not mentioned). For [1p] using `@fluidframework` we would want to note that /legacy should be treated as unstable (can change in minor) until we note otherwise.

(I don't like making it complicated, but it is a bit :( )

kashms replied:
This is for general external guidance, I think for 1P we should have publish that somewhere internally to not complicate the external messaging. But I'll update the pre-condition to include the fluid-framework package.

kashms committed:
Added more text for clarity

DLehenbauer commented on docs/content/docs/build/releases-and-apitags.md:
```diff
@@ -37,10 +37,20 @@ There are no guarantees for API breakages or Fluid document compatibility among
--   **Internal APIs** (`@internal`) - Internal Fluid APIs, should not be used by applications.
+For packages that are part of the `@fluidframework` scope and the `fluid-framework` package, we use import paths to communicate the stability and guarantees associated with those APIs.
+
+-   **Public APIs** - These APIs are officially supported and any breaking changes to these APIs will be done in a major release and go through a deprecation phase before being removed. We guarantee to not break them in minor and patch releases (consistent with the semver guarantees)
```
I would remove the 'we guarantee" sentence.

DLehenbauer commented:
@kashms - I suggest the following changes:

* Move discussion of experimental/internal packages to end, as these are relevant for the smallest audience.
* Removed "We guarantee..." from public, since minor/patch breaks could happen unintentionally (or we could change our policy in the future.)
* Removed "Rest assured..." from legacy, since plans could change (e.g., we find it more economical to support legacy DDSes indefinitely.)

(#21490 is an optional PR into your branch that makes these changes with some other light editing.)

DLehenbauer committed:
Update releases-and-apitags.md (#21490)

Suggested changes:
- Move discussion of experimental/internal packages to end, as these are
relevant for the smallest audience.
- Removed "We guarantee..." from public, since minor/patch breaks could
happen unintentionally (or we could change our policy in the future.)
- Removed "Rest assured..." from legacy, since it's possible we will
support legacy DDSes indefinitely.

kashms committed:
Merge branch 'main' into kashms-patch-1
